ui/Url.elm

```diff
@@ -0,0 +1,204 @@
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = "all"
+                    , platformFilter = "all"
+                    , versionFilter = "all"
+                    , localeFilter = "all"
+                }
+
+            Just (ChannelView product channel) ->
+                { model
+                    | route = ChannelView product channel
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = "all"
+                    , versionFilter = "all"
+                    , localeFilter = "all"
+                }
+
+            Just (PlatformView product channel platform) ->
+                { model
+                    | route = PlatformView product channel platform
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = Debug.log "platform filter" platform
+                    , versionFilter = "all"
+                    , localeFilter = "all"
+                }
+
+            Just (VersionView product channel platform version) ->
+                { model
+                    | route = VersionView product channel platform version
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = Debug.log "platform filter" platform
+                    , versionFilter = Debug.log "version filter" version
+                    , localeFilter = "all"
+                }
+
+            Just (LocaleView product channel platform version locale) ->
+                { model
+                    | route = VersionView product channel platform version
+                    , buildIdFilter = ""
+                    , productFilter = Debug.log "product filter" product
+                    , channelFilter = Debug.log "channel filter" channel
+                    , platformFilter = Debug.log "platform filter" platform
+                    , versionFilter = Debug.log "version filter" version
+                    , localeFilter = Debug.log "locale fitler" locale
+                }
+
+            _ ->
+                { model
+                    | route = MainView
+                    , buildIdFilter = ""
+                    , productFilter = "all"
+                    , channelFilter = "all"
+                    , platformFilter = "all"
+                    , versionFilter = "all"
+                    , localeFilter = "all"
+                }
+
+
+urlFromRoute : Model -> String
+urlFromRoute model =
+    case model.route of
```

Comment:
Nit: If we're only using the `route` field, it sounds faire to just accept a `Route` argument, not a full `Model` one.